test_code_skb.js

```diff
@@ -13,33 +13,58 @@ var sample = 1000;
 var R = 8.3145; // J/mol*K
 var T = 273;    // K
 
+function plotFunction(functionToPlot, xDomain, yRange, xScreenBoundary, yScreenBoundary) {
+    var numSamples = 1000
+    var x = d3.scale.linear().domain(xDomain).range(xScreenBoundary);
+    var y = d3.scale.linear().domain(yRange).range(yScreenBoundary);
+
+    var x1 = d3.scale.linear().domain([0, sample]).range(xDomain);
+    var data = d3.range(numSamples).map(function (d) {
+        return {
+            x: x1(d),
+            y: functionToPlot(x1(d))
+        };
+    });
+    var line = d3.svg.line()
+        .x(function (d) {
+            return x(d.x);
+        })
+        .y(function (d) {
+            return y(d.y);
+        });
+    return line(data);
+}
+
```

Comment:
I mostly copy-pasted the functions that were duplicated and needed for plotting
Before the yRange was auto-determined. I think you always want your yRange and yScreenBoundary to be the same for plots going on the same graph. This way scale wont change? I didn't test that out though...